lib/commands/recordscreen.js

```diff
@@ -237,9 +239,11 @@ commands.startRecordingScreen = async function startRecordingScreen (options = {
     suffix: MP4_EXT,
   });
 
+  const wdaBaseUrl = this.opts.wdaBaseUrl || WDA_BASE_URL;
   const screenRecorder = new ScreenRecorder(this.opts.device.udid, videoPath, {
     remotePort: this.opts.mjpegServerPort || DEFAULT_MJPEG_SERVER_PORT,
-    usePortForwarding: this.isRealDevice(),
+    remoteUrl: wdaBaseUrl,
+    usePortForwarding: this.isRealDevice() && [WDA_BASE_URL, WDA_BASE_URL_IP].includes(wdaBaseUrl),
```

Comment:
I'd rather extract the actual host and check for something like
```
host === "127.0.0.1" || host === "::ffff:127.0.0.1" || host === "::1" || host.includes("localhost");
```

Reply:
this check might also be extracted into utilities module, since it's used more than once